CMakeLists.txt

```diff
@@ -45,12 +45,19 @@ add_custom_target(linkp4cxdp
   COMMAND ${CMAKE_COMMAND} -E create_symlink ${P4C_BINARY_DIR}/p4include ${CMAKE_CURRENT_BINARY_DIR}/p4include
   )
 
+set (XFAIL_TESTS_XDP
+  # Fails because of faulty syntax
+  ${CMAKE_CURRENT_SOURCE_DIR}/tests/xdp.p4
+  # Fails because of stack size
+  ${CMAKE_CURRENT_SOURCE_DIR}/tests/xdp7.p4
+  ${CMAKE_CURRENT_SOURCE_DIR}/tests/xdp12.p4
+  )
+
 add_dependencies(p4c_driver linkp4cxdp)
 
-set(XDP_DRIVER ${CMAKE_CURRENT_SOURCE_DIR}/run-p4c-xdp.sh)
+set(XDP_DRIVER "${CMAKE_CURRENT_SOURCE_DIR}/run-ebpf-test.py -t xdp")
+
 
 # This file will not run the full tests, but it will attempt to compile the p4 files down to C
-set (XDP_TEST_SUITES
-  "${CMAKE_CURRENT_SOURCE_DIR}/tests/xdp*.p4"
-  )
-p4c_add_tests("xdp" ${XDP_DRIVER} ${XDP_TEST_SUITES} "")
+set (XDP_TEST_SUITES "${CMAKE_CURRENT_SOURCE_DIR}/tests/xdp*.p4")
+p4c_add_tests("xdp" ${XDP_DRIVER} ${XDP_TEST_SUITES} "${XFAIL_TESTS_XDP}")
```

Comment:
Adding the xfail tests does not work for some reason, can't figure out why...

Reply:
Is this problem solved?

Reply:
No, unfortunately not. I will look into it next week. I also do not know what to do about the travis build, it might be because travis is built with an Ubuntu 14.04 image.

Reply:
The problem is in the p4c_add_tests function. It resolves the path of normal but not the xfail tests. As a result, the path of the xfail test does not match up with the given list and is ignored.